pytorch_lightning/trainer/trainer.py

```diff
@@ -992,12 +997,45 @@ def _run_train(self) -> None:
                 self.state.stage = None
         except BaseException:
             self.state.status = TrainerStatus.INTERRUPTED
+            # try syncing remaing processes, kill otherwise
+            self._syncing_processes(traceback.format_exc())
             # give accelerators a chance to finish
             self.accelerator.on_train_end()
             # reset bookkeeping
             self.state.stage = None
             raise
 
+    def _syncing_processes(self, trace: str):
+        if distributed_available() and isinstance(self.accelerator.training_type_plugin, DDPPlugin):
+            sync_dir = os.getenv("PL_TMPDIR")
+
+            if not os.path.exists(sync_dir):
+                # avoid race condition
+                try:
+                    os.makedirs(sync_dir)
+                except FileExistsError:
+                    pass
```

Comment:
you can move this to DDP init, since this can be done before the procs are initialized (then you also don't have race-conditions)

Reply:
plus `exists_ok=True` would probably work too?

Reply:
@awaelchli Not necessarily. This would check in the beginning and then create which could lead to race conditions in some cases where all processes are created simultaneously

Reply:
yes, so the condition `if not os.path.exists(sync_dir)` does not matter, as it is handled by the try-except below anyway?

Reply:
You definitely need the try-except but you can merge the condition

Reply:
Done !